pkg/virt-controller/watch/pool/pool.go

```diff
@@ -1650,3 +1657,40 @@ func (c *Controller) handlePoolDeletion(pool *poolv1.VirtualMachinePool, vms []*
 
 	return nil
 }
+
+func isAutohealingEnabled(pool *poolv1.VirtualMachinePool) bool {
+	return pool.Spec.Autohealing != nil && *pool.Spec.Autohealing
+}
+
+func (c *Controller) autoHealFailingVMs(pool *poolv1.VirtualMachinePool, vms []*virtv1.VirtualMachine) error {
+	vmsToCleanup := filterVMs(vms, func(vm *virtv1.VirtualMachine) bool {
+		return vm.Status.StartFailure != nil && vm.Status.StartFailure.ConsecutiveFailCount >= defaultStartUpFailureThreshold
```

Comment:
Are there more conditions to check if the VM is failing? i.e. not schedulable, failed and stopped, etc.